### DISCUSSION TITLE
Error at update to 20220808 (still present in rAudio • 20220828)

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#976] [Fixed Issues]
FWIW
Update to 20220808 generates error messages:
<img src="https://user-images.githubusercontent.com/25383513/183467097-130e3b93-6813-4994-bea5-ece840f1d730.png" width=600>

**Comment by Raitsa** [comment]
No errors here, updated two raspberry 4 players today 8.8.2022.

**Comment by klassikmann** [comment]
Update to 20220813, error messages still present:
<img src="https://user-images.githubusercontent.com/25383513/184501677-68528f11-30c4-4dd0-91eb-57c4a9311555.jpg" width=600>

**Comment by ceejayemm** [comment]
@klassikmann 
No problem here, I am afraid, for me updating 2 x RPi (3A and 3B Plus) to 20220813. 

RPi3A does not have a DAC, only uses Bluetooth output.  RPi3B+ uses an IQaudIO DAC Pro. Does this help ?  It seems the source file you reference is trying to match the DAC type to a driver (?) type ? May be your DAC doesn't have a driver listed in this file ? - only guessing here.

Chris

**Reply by klassikmann** [reply]
@ceejayemm 
Despite the error messages my installation is up and running. I cannot detect any irregularities.

I don't think the file system-i2s.json is the culprit, it's only the last in the list of the files to be updated (cf. the screenshot in my original posting). I assume the origin of the error messages lies in the various bash-scripts of rAudio invoking "sed" and "convert" (part of the imagemagick package).

btw.: I'm using no DAC but an digital-out (HiFiBerry Digi+ Pro) connected to the digital-in of my active speakers.

**Comment by hjfisch** [comment]
I've 2 RPI 4 with Allo Digi One build into my 2 pairs of active speakers and no probs with updating to the last build .

**Comment by klassikmann** [comment]
Update to 20220814:
Despite the update notes the error messages persist:

> "Fix - Bookmarks rename/remove"
> "Fix - Web Radio data refresh on start playing"
> ""Improve - Bookmark thumbnails"

My latest assumption was that an update is trying to remediate existing bookmarks.

**Comment by klassikmann** [comment]
rAudio • 20220828:
Error messages as presented in first post still present. 

@rern 
I'm prepared to investigate the cause, but I need a hint where to search for.

**Comment by R101G** [comment]
I get the same errors, but have not experienced any problems. I assumed they were associated with a feature I do not use.

**Comment by rern** [comment]
@klassikmann,

Please try:
```sh
cmd.sh dirpermissions
```
If it showed the same errors, try each line of the script:
 ```sh
chmod 755 /srv /srv/http /srv/http/* /mnt /mnt/MPD /mnt/MPD/*/
chown http:http /srv /srv/http /srv/http/* /mnt /mnt/MPD /mnt/MPD/*/
chmod -R 755 /srv/http/{assets,bash,data,settings}
chown -R http:http /srv/http/{assets,bash,data,settings}
chown mpd:audio /srv/http/data/mpd /srv/http/data/mpd.db /srv/http/data/playlists
```

**Reply by klassikmann** [reply]
@rern 
> Please try:
> cmd.sh dirpermissions

No error messages, all permissions o.k.

btw. The last line of your script should read:
`chown mpd:audio /srv/http/data/mpd /srv/http/data/mpd/mpd.db /srv/http/data/playlists`

**Comment by ceejayemm** [comment]
For info - just updated my 3 x RPi (all 64 bit) to 20220828 with no issues

RPi3B+ (Kernel: 5.10.82-4)
RPi3B (Kernel: 5.15.40-3) 
RPi3A (Kernel: 5.15.40-3)

Chris

**Reply by AndyFx12** [reply]
On my system, error is produced by 
`cmd.sh color`
So I think, only users with changed colors will be affected?!

Regards, Andy

**Reply by R101G** [reply]
Just to confirm - I do have a changed colour.

**Reply by klassikmann** [reply]
> On my system, error is produced by `cmd.sh color` So I think, only users with changed colors will be affected?!
> 
> Regards, Andy

Interesting find, Andy. I'm using a custom color. Unfortunately, I have no access to my installation now and for the next two weeks.

**Reply by AndyFx12** [reply]
After some digging into cmd.sh, color:

` s|(.box{fill:hsl).*|\1($hsl);|
s|(.text{fill:hsl).*|\1(${hsg}30%);}|
" $dirimg/coverart.svg
	sed -i -E "
s|(.box{fill:hsl).*|\1($hsl);}|
s|(path{fill:hsl).*|\1(${hsg}75%);}|
" $dirimg/icon.svg
	sed -E "s|(path{fill:hsl).*|\1(0,0%,90%);}|" $dirimg/icon.svg \`

I think there are some backslash´s missing in front of "{":
` s|(.box\{fill:hsl).*|\1($hsl);|
s|(.text\{fill:hsl).*|\1(${hsg}30%);}|
" $dirimg/coverart.svg
	sed -i -E "
s|(.box\{fill:hsl).*|\1($hsl);}|
s|(path\{fill:hsl).*|\1(${hsg}75%);}|
" $dirimg/icon.svg
	sed -E "s|(path\{fill:hsl).*|\1(0,0%,90%);}|" $dirimg/icon.svg \`

After changing, error is gone. 

Issue don´t produce icons with selected color - so no big thing.

Andy

**Comment by AndyFx12** [comment]
Update branch fixed this error for me.
Andy

**Reply by klassikmann** [reply]
Back at home, upgrade to rAudio • 20220909: Problem fixed.
